src/pages/Typing/index.tsx

```diff
@@ -168,67 +136,27 @@ const App: React.FC = () => {
       }
 
       mixpanel.track('Chapter', chapterStatUpload)
-
-      setIsChapterEnd(true)
-    } else {
-      setOrder((order) => order + 1)
     }
-
-    // if user finished the word without skipping, then set skipState to false
-    setIsShowSkip(false)
   }
 
-  const addChapter = useCallback(() => {
-    if (wordList === undefined) {
-      return
+  useEffect(() => {
+    // 启动计时器
+    function tick() {
+      dispatch({ type: TypingStateActionType.TICK_TIMER })
+      requestAnimationFrameRef.current = requestAnimationFrame(tick)
```

Comment:
Can we use `setInterval` instead? This affects performance considerably because `requestAnimationFrame` is called 60 times each second. Here is the flame chart from the performance debugging panel.

- This branch:
  <img width="400" alt="image" src="https://user-images.githubusercontent.com/2239547/228591942-8679c446-68d5-4eb0-b062-bb198f8f4302.png">
- On the main branch:
  <img width="400" alt="image" src="https://user-images.githubusercontent.com/2239547/228591765-e7cd099c-a7d9-4ece-bb7f-b7b283eb0080.png">

As you can see there are too many redundant function calls.